frontend/src/views/positions/positions-list-toolbar.js

```diff
@@ -0,0 +1,148 @@
+import React from "react";
+import { formatColumnName } from "../../libs/utils";
+import { ConnectedAddPositionDialog } from "./add-position-dialog";
+import {
+    Badge,
+    Button,
+    ButtonToolbar,
+    ToggleButton,
+    ToggleButtonGroup,
+} from "react-bootstrap";
+
+const BLACKLIST = ["instructor_preferences"]; // Any fields here will not be accessible in the column selector.
+
+/**
+ * Builds an array (with no duplicates) of all column headings which appear at least once in the positions data from the database
+ * All headings which are present in the blacklist (defined above) are removed from this list
+ *
+ * @param {*} positions the positions data which was returned from a GET request to the admin/positions endpoint.
+ * @returns an array with no duplicates containing headings in the format [a-z][a-z]*_[a-z]*
+ */
+function getFetchedColumnHeadings(positions) {
+    return [
+        ...new Set(
+            positions
+                .reduce(
+                    (acc, position) => [...acc, ...Object.keys(position)],
+                    []
+                )
+                .filter((heading) => !BLACKLIST.includes(heading))
+        ),
+    ];
+}
+
+/**
+ * Builds an array of ToggleButton elements for use in the column selector component.
+ *
+ * @param {*} viewableColumns an array of all viewable column names imported from the database in the form [a-z][a-z]*_[a-z]*
+ * @param {*} selectedColumns a set of the the column names which are selected (currently being displayed)
+ */
+function getColumnSelector(viewableColumns, selectedColumns) {
+    return viewableColumns.map((heading) => {
+        const lowercase = formatColumnName(heading).toLowerCase();
+        const isSelected = selectedColumns.has(lowercase);
+
+        return (
+            <ToggleButton
+                type="checkbox"
+                value={lowercase}
+                key={lowercase}
+                variant="light"
+                size="sm"
+                className="text-left font-weight-lighter text-capitalize text-nowrap flex-grow-0 m-1"
+            >
+                {formatColumnName(heading)}
+                <Badge
+                    variant={
+                        isSelected ? "outline-primary" : "outline-secondary"
+                    }
+                >
+                    {isSelected ? "✓" : "❌"}
+                </Badge>
+            </ToggleButton>
+        );
+    });
+}
```

Comment:
We don't want dynamic setting of columns. A "full information" button that opens a dialog with all the information would be good though. Such a dialog should tap into a new view made especially for showing/editing information about a single position.